### ISSUE TITLE
`backup now` command always got invalid item name.

### ISSUE THREAD

**Issue by metrue** [Azure/azure-cli#10409]
> ### `az feedback` auto-generates most of the information requested below, as of CLI 
```
 az -v
azure-cli                         2.0.71 *

command-modules-nspkg               2.0.3
core                              2.0.71 *
nspkg                              3.0.4
telemetry                          1.0.3

Python location '/usr/local/Cellar/azure-cli/2.0.71/libexec/bin/python'
Extensions directory '/Users/huangminghe/.azure/cliextensions'

Python (Darwin) 3.7.4 (default, Jul  9 2019, 18:13:23)
[Clang 10.0.1 (clang-1001.0.46.4)]

Legal docs and information: aka.ms/AzureCliLegal
```

**Describe the bug**

Always got error 
```
The command failed with an unexpected error. Here is the traceback:

Item not found. Please provide a valid item_name.
Traceback (most recent call last):
```

But actually the item name is got from `az list` command, it should not be wrong.
```
 az backup item list --resource-group Minghe_test --vault-name backupvaultMinghetestdemo
Command group 'backup' is in preview. It may be changed/removed in a future release.
[
  {
    "eTag": null,
    "id": "xxxx",
    "location": null,
    "name": "VM;iaasvmcontainerv2;Minghe_test;democ",
    "properties": {
      "backupManagementType": "AzureIaasVM",
      "backupSetName": null,
      "containerName": "iaasvmcontainerv2;Minghe_test;democ",
      "createMode": null,
      "extendedInfo": null,
      "friendlyName": "democ",
      "healthDetails": null,
      "healthStatus": "Passed",
      "lastBackupStatus": "",
      "lastBackupTime": "2001-01-01T00:00:00+00:00",
      "lastRecoveryPoint": null,
      "policyId": "xxxx",
      "policyName": "DefaultPolicy",
      "protectedItemDataId": "35186015689227",
      "protectedItemType": "Microsoft.Compute/virtualMachines",
      "protectionState": "IRPending",
      "protectionStatus": "Healthy",
      "sourceResourceId": "xxx",
      "virtualMachineId": "xxxx",
      "workloadType": "VM"
    },
    "resourceGroup": "Minghe_test",
    "tags": null,
    "type": "Microsoft.RecoveryServices/vaults/backupFabrics/protectionContainers/protectedItems"
  }
]
```
Then I tried to run `backup now` with
```
 az backup protection backup-now -g Minghe_test -c "iaasvmcontainer;iaasvmcontainerv2;Minghe_test;democ" -i "VM;iaasvmcontainerv2;Minghe_test;democ" --retain-until '01-10-2019' --vault-name backupvaultMinghetestdemoa
Command group 'backup' is in preview. It may be changed/removed in a future release.
The command failed with an unexpected error. Here is the traceback:

Item not found. Please provide a valid item_name.
Traceback (most recent call last):
  File "/usr/local/Cellar/azure-cli/2.0.71/libexec/lib/python3.7/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/local/Cellar/azure-cli/2.0.71/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 603, in execute
    raise ex
  File "/usr/local/Cellar/azure-cli/2.0.71/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 661, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/usr/local/Cellar/azure-cli/2.0.71/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 654, in _run_job
    six.reraise(*sys.exc_info())
  File "/usr/local/Cellar/azure-cli/2.0.71/libexec/lib/python3.7/site-packages/six.py", line 693, in reraise
    raise value
  File "/usr/local/Cellar/azure-cli/2.0.71/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 631, in _run_job
    result = cmd_copy(params)
  File "/usr/local/Cellar/azure-cli/2.0.71/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 305, in __call__
    return self.handler(*args, **kwargs)
  File "/usr/local/Cellar/azure-cli/2.0.71/libexec/lib/python3.7/site-packages/azure/cli/core/__init__.py", line 485, in default_command_handler
    return op(**command_args)
  File "/usr/local/Cellar/azure-cli/2.0.71/libexec/lib/python3.7/site-packages/azure/cli/command_modules/backup/custom.py", line 262, in backup_now
    _validate_item(item)
  File "/usr/local/Cellar/azure-cli/2.0.71/libexec/lib/python3.7/site-packages/azure/cli/command_modules/backup/custom.py", line 696, in _validate_item
    _validate_object(item, "Item not found. Please provide a valid item_name.")
  File "/usr/local/Cellar/azure-cli/2.0.71/libexec/lib/python3.7/site-packages/azure/cli/command_modules/backup/custom.py", line 705, in _validate_object
    raise ValueError(error_message)
ValueError: Item not found. Please provide a valid item_name.

To open an issue, please run: 'az feedback'
```

**To Reproduce**
Above.

**Expected behavior**
The item name is got from `az list` command, it cannot be incorrect.

**Environment summary**
```
uname -a
Darwin MinghedeMacBook-Pro.local 18.6.0 Darwin Kernel Version 18.6.0: Thu Apr 25 23:16:27 PDT 2019; root:xnu-4903.261.4~2/RELEASE_X86_64 x86_64
```
**Additional context**

**Comment by pvrk** [comment]
Thanks for the feedback. Now the issue is clear. If you run the az backup item list command with output as table, there will be a column called 'Name'. That should be passed as name. The value under that column 'Name' is actually 'FriendlyName' in the JSON output. Basically, you should pass the value 'FriendlyName' as 'Name'. We wanted users to pass the friendlyName but as per CLI guidelines the parameter had to '-Name'. Hope this is clear. This can be clarified with clear documentation and examples. I will start working as soon as I can.

**Comment by pvrk** [comment]
We have fixed this in the Jan, 2020 release but somhow I missed updating this thread. My sincere apologies.
@metrue : Your item list command and using name is correct. The only requirement is that also use the ```az backup container list``` command to get the name for container as well. If you list container and take the name parameter and list the item and take the name parameter, then any combination of item and container would work.

For eg: Your command that you are attempting is this:

```az backup protection backup-now -g Minghe_test -c "iaasvmcontainer;iaasvmcontainerv2;Minghe_test;democ" -i "VM;iaasvmcontainerv2;Minghe_test;democ" --retain-until '01-10-2019' --vault-name backupvaultMinghetestdemoa```

What you have to try is this:
```az backup container list --resource-group Minghe_test --vault-name backupvaultMinghetestdemo --backup-management-type AzureIaaSVM```

[
  {
    "eTag": null,
    "id": "xxxx",
    "location": null,
    "name": "IaasVMContainer;iaasvmcontainerv2;Minghe_test;democ",
    "properties": {
      "backupManagementType": "AzureIaasVM",

.
.
.
]

```az backup protection backup-now -g Minghe_test -c "IaasVMContainer;iaasvmcontainerv2;;Minghe_test;democ" -i "VM;iaasvmcontainerv2;Minghe_test;democ" --retain-until '01-10-2019' --vault-name backupvaultMinghetestdemoa```

This would work. We have verified this in internal testing too.

To summarize, you can use the "name" parameter of the JSON output for unique identification. For container identification, use container list/show command and for backup item identification, use backup item list/show command.

Sincere apologies for the confusion caused by my earlier comments. I will make sure that this doesn't get repeated again.

Hope this solves the issue.

**Comment by pvrk** [comment]
Closing this thread since the fix is provided. Please feel free to re-open if this doesn't solve the issue.
